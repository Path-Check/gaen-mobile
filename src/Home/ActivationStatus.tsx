import React, { FunctionComponent } from "react"
import { useTranslation } from "react-i18next"
import { TouchableOpacity, View, StyleSheet } from "react-native"
import { SvgXml } from "react-native-svg"

import { Text } from "../components"

import { Icons } from "../assets"
import { Colors, Iconography, Outlines, Spacing, Typography } from "../styles"

type Content = {
  bodyText: string
  leftIcon: string
  leftIconFill: string
  rightIcon: string
  onPress: () => void
  accessibilityLabel: string
}

interface ActivationStatusProps {
  headerText: string
  isActive: boolean
  infoAction: () => void
  fixAction: () => void
  testID: string
}

export const ActivationStatus: FunctionComponent<ActivationStatusProps> = ({
  headerText,
  isActive,
  infoAction,
  fixAction,
  testID,
}) => {
  const { t } = useTranslation()

  const activeContent: Content = {
    bodyText: t("common.enabled"),
    leftIcon: Icons.CheckInCircle,
    leftIconFill: Colors.success100,
    rightIcon: Icons.HomeInfo,
    onPress: infoAction,
    accessibilityLabel: t("home.get_more_info", { technology: headerText }),
  }

  const inactiveContent: Content = {
    bodyText: t("common.disabled"),
    leftIcon: Icons.XInCircle,
    leftIconFill: Colors.danger75,
    rightIcon: Icons.Wrench,
    onPress: fixAction,
    accessibilityLabel: t("home.fix", { technology: headerText }),
  }

  const content = isActive ? activeContent : inactiveContent

  return (
    <TouchableOpacity
      onPress={content.onPress}
      style={style.activationStatusContainer}
      accessible
      accessibilityLabel={content.accessibilityLabel}
      testID={testID}
    >
      <View style={style.activationStatusLeftContainer}>
        <SvgXml
          xml={content.leftIcon}
          fill={content.leftIconFill}
          width={Iconography.small}
          height={Iconography.small}
        />
        <View style={style.activationStatusTextContainer}>
          <Text style={style.bottomHeaderText}>{headerText}</Text>
          <Text style={style.bottomBodyText}>{content.bodyText}</Text>
        </View>
      </View>
      <View style={style.activationStatusRightContainer}>
        <SvgXml xml={content.rightIcon} />
      </View>
    </TouchableOpacity>
  )
}

const rightColumnWidth = 70
const style = StyleSheet.create({
  activationStatusContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingVertical: Spacing.xSmall,
    marginHorizontal: Spacing.small,
    borderBottomWidth: Outlines.hairline,
    borderBottomColor: Colors.neutral10,
  },
  activationStatusLeftContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  activationStatusTextContainer: {
    marginLeft: Spacing.medium,
  },
  activationStatusRightContainer: {
    width: rightColumnWidth,
    alignItems: "center",
  },
  bottomHeaderText: {
    ...Typography.body1,
    color: Colors.black,
    lineHeight: Typography.smallLineHeight,
  },
  bottomBodyText: {
    ...Typography.body2,
    color: Colors.neutral75,
    lineHeight: Typography.xSmallLineHeight,
  },
})
