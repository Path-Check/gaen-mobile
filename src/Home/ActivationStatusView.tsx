import React, { FunctionComponent } from "react"
import { TouchableOpacity, View, StyleSheet } from "react-native"
import { SvgXml } from "react-native-svg"
import { useTranslation } from "react-i18next"

import { Text } from "../components"

import { Icons } from "../assets"
import {
  Affordances,
  Colors,
  Iconography,
  Outlines,
  Spacing,
  Typography,
} from "../styles"

type Content = {
  backgroundColor: string
  borderColor: string
  bodyText: string
  statusIcon: string
  actionText: string
  onPress: () => void
  accessibilityLabel: string
  chevron: string
}

interface ActivationStatusProps {
  headerText: string
  isActive: boolean
  infoAction: () => void
  fixAction: () => void
  testID: string
}

const ActivationStatusView: FunctionComponent<ActivationStatusProps> = ({
  headerText,
  isActive,
  infoAction,
  fixAction,
  testID,
}) => {
  const { t } = useTranslation()

  const activeContent: Content = {
    backgroundColor: Colors.accent.success25,
    borderColor: Colors.accent.success100,
    bodyText: t("common.on"),
    statusIcon: Icons.CheckInCircle,
    actionText: t("exposure_scanning_status.learn_more"),
    onPress: infoAction,
    accessibilityLabel: t("home.get_more_info", { technology: headerText }),
    chevron: Icons.ChevronUp,
  }

  const inactiveContent: Content = {
    backgroundColor: Colors.accent.danger25,
    borderColor: Colors.accent.danger100,
    bodyText: t("common.off"),
    statusIcon: Icons.XInCircle,
    actionText: t("exposure_scanning_status.fix_this"),
    onPress: fixAction,
    accessibilityLabel: t("home.fix", { technology: headerText }),
    chevron: Icons.ChevronRight,
  }

  const content = isActive ? activeContent : inactiveContent

  const {
    backgroundColor,
    borderColor,
    bodyText,
    statusIcon,
    actionText,
    onPress,
    accessibilityLabel,
    chevron,
  } = content

  const outerContainerStyle = {
    ...style.outerContainer,
    backgroundColor: backgroundColor,
    borderColor: borderColor,
  }

  const statusTextContainerStyle = {
    ...style.statusTextContainer,
    backgroundColor: borderColor,
  }

  return (
    <TouchableOpacity
      onPress={onPress}
      style={outerContainerStyle}
      accessible
      accessibilityLabel={accessibilityLabel}
      testID={testID}
    >
      <View style={style.topContainer}>
        <View style={style.topTextContainer}>
          <Text style={style.systemServiceText}>{headerText}</Text>
          <View style={statusTextContainerStyle}>
            <Text style={style.statusText}>{bodyText}</Text>
          </View>
        </View>
        <SvgXml
          xml={statusIcon}
          fill={borderColor}
          width={Iconography.medium}
          height={Iconography.medium}
        />
      </View>
      <View style={style.bottomContainer}>
        <Text style={style.actionText}>{actionText}</Text>
        <SvgXml
          xml={chevron}
          fill={Colors.neutral.black}
          width={Iconography.tiny}
          height={Iconography.tiny}
        />
      </View>
    </TouchableOpacity>
  )
}

const style = StyleSheet.create({
  outerContainer: {
    ...Affordances.floatingContainer,
    borderWidth: Outlines.thin,
  },
  topContainer: {
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "space-between",
    marginBottom: Spacing.small,
  },
  topTextContainer: {
    marginTop: Spacing.xxSmall,
    borderColor: Colors.neutral.shade10,
  },
  systemServiceText: {
    ...Typography.header.x40,
    color: Colors.neutral.black,
    lineHeight: Typography.lineHeight.x30,
    marginBottom: Spacing.xSmall,
  },
  statusTextContainer: {
    alignSelf: "flex-start",
    paddingVertical: Spacing.xxxSmall,
    paddingHorizontal: Spacing.xxSmall,
    borderRadius: Outlines.baseBorderRadius,
  },
  statusText: {
    ...Typography.body.x20,
    ...Typography.style.bold,
    color: Colors.neutral.white,
    lineHeight: Typography.lineHeight.x20,
  },
  bottomContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  actionText: {
    ...Typography.body.x30,
    color: Colors.neutral.black,
    marginRight: Spacing.xxSmall,
    paddingBottom: 2,
  },
})

export default ActivationStatusView
