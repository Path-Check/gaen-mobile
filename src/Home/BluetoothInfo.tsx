import React, { FunctionComponent } from "react"
import {
  ScrollView,
  TouchableOpacity,
  View,
  Image,
  StyleSheet,
} from "react-native"
import { SvgXml } from "react-native-svg"
import { useTranslation } from "react-i18next"
import { useNavigation } from "@react-navigation/native"

import { GlobalText } from "../components/GlobalText"

import { Layout, Typography, Spacing, Colors, Iconography } from "../styles"
import { Icons, Images } from "../assets"

const BluetoothInfo: FunctionComponent = () => {
  const navigation = useNavigation()
  const { t } = useTranslation()

  return (
    <View style={style.container}>
      <View style={style.headerContainer}>
        <GlobalText style={style.headerText}>
          {t("home.bluetooth_info_header")}
        </GlobalText>
        <TouchableOpacity
          style={style.closeIconContainer}
          onPress={navigation.goBack}
        >
          <SvgXml
            xml={Icons.XInCircle}
            fill={Colors.neutral30}
            width={Iconography.small}
            height={Iconography.small}
          />
        </TouchableOpacity>
      </View>
      <ScrollView
        contentContainerStyle={style.mainContentContainer}
        alwaysBounceVertical={false}
      >
        <View style={style.textContainer}>
          <GlobalText style={style.bodyText}>
            {t("home.bluetooth_info_body")}
          </GlobalText>
        </View>
        <Image source={Images.PhonesSharingCodes} style={style.image} />
      </ScrollView>
    </View>
  )
}

const headerHeight = 70

const style = StyleSheet.create({
  container: {
    flex: 1,
    height: "100%",
    backgroundColor: Colors.primaryLightBackground,
  },
  headerContainer: {
    position: "absolute",
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-between",
    backgroundColor: Colors.secondary10,
    zIndex: Layout.zLevel1,
    height: headerHeight,
  },
  headerText: {
    flex: 10,
    ...Typography.header3,
    paddingVertical: Spacing.medium,
    paddingHorizontal: Spacing.large,
    color: Colors.primary125,
  },
  closeIconContainer: {
    flex: 1,
    alignItems: "flex-end",
    padding: Spacing.medium,
  },
  mainContentContainer: {
    paddingTop: headerHeight + Spacing.large,
    paddingBottom: Spacing.huge,
  },
  image: {
    width: "100%",
    height: 290,
    resizeMode: "contain",
  },
  textContainer: {
    paddingHorizontal: Spacing.large,
    marginBottom: Spacing.small,
  },
  bodyText: {
    ...Typography.mainContent,
  },
})
export default BluetoothInfo
