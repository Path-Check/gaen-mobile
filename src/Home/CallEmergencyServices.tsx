import React, { FunctionComponent } from "react"
import { StyleSheet, TouchableOpacity, Linking } from "react-native"
import { SvgXml } from "react-native-svg"
import { useTranslation } from "react-i18next"

import { Text } from "../components"

import {
  Iconography,
  Spacing,
  Buttons,
  Typography,
  Colors,
  Outlines,
} from "../styles"
import { Icons } from "../assets"

interface CallEmergencyServicesProps {
  phoneNumber: string
}

const CallEmergencyServices: FunctionComponent<CallEmergencyServicesProps> = ({
  phoneNumber,
}) => {
  const { t } = useTranslation()
  const handleOnPressCallEmergencyServices = () => {
    Linking.openURL(`tel:${phoneNumber}`)
  }

  const buttonText = t("home.call_emergency_services")

  return (
    <TouchableOpacity
      onPress={handleOnPressCallEmergencyServices}
      accessibilityLabel={buttonText}
      accessibilityRole="button"
      style={style.emergencyButtonContainer}
    >
      <SvgXml
        xml={Icons.Phone}
        fill={Colors.neutral.white}
        width={Iconography.xSmall}
        height={Iconography.xSmall}
      />
      <Text style={style.emergencyButtonText}>{buttonText}</Text>
    </TouchableOpacity>
  )
}

const style = StyleSheet.create({
  emergencyButtonContainer: {
    ...Buttons.thin.base,
    borderRadius: Outlines.borderRadiusLarge,
    backgroundColor: Colors.accent.danger100,
    padding: Spacing.large,
  },
  emergencyButtonText: {
    ...Typography.button.primary,
    marginLeft: Spacing.small,
  },
})

export default CallEmergencyServices
