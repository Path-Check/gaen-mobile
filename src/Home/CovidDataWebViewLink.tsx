import React, { FunctionComponent } from "react"
import { Pressable, View, StyleSheet } from "react-native"
import { useTranslation } from "react-i18next"
import { useNavigation } from "@react-navigation/native"
import { SvgXml } from "react-native-svg"

import { Text } from "../components"
import { ModalStackScreens } from "../navigation"

import { Icons } from "../assets"
import {
  Spacing,
  Colors,
  Typography,
  Outlines,
  Iconography,
  Affordances,
} from "../styles"

const CovidDataWebViewLink: FunctionComponent = () => {
  const { t } = useTranslation()
  const navigation = useNavigation()

  const handleOnPress = async () => {
    navigation.navigate(ModalStackScreens.CovidDataWebView)
  }

  return (
    <Pressable
      style={style.container}
      onPress={handleOnPress}
      accessibilityLabel={t("home.covid_data")}
    >
      <SvgXml
        xml={Icons.BarGraph}
        fill={Colors.neutral.shade75}
        width={Iconography.xxSmall}
        height={Iconography.xxSmall}
      />

      <View style={style.textContainer}>
        <Text style={style.shareText}>{t("home.covid_data")}</Text>
      </View>
      <SvgXml
        xml={Icons.ChevronRight}
        fill={Colors.neutral.shade75}
        width={Iconography.xxSmall}
        height={Iconography.xxSmall}
      />
    </Pressable>
  )
}

const style = StyleSheet.create({
  container: {
    ...Affordances.floatingContainer,
    paddingVertical: Spacing.small,
    flexDirection: "row",
    alignItems: "center",
    borderColor: Colors.primary.shade100,
    borderWidth: Outlines.thin,
  },
  textContainer: {
    flex: 1,
    marginLeft: Spacing.medium,
  },
  shareText: {
    ...Typography.body.x30,
    ...Typography.style.medium,
    color: Colors.text.primary,
  },
})

export default CovidDataWebViewLink
