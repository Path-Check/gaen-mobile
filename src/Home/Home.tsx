import React from "react"
import {
  StyleSheet,
  TouchableOpacity,
  View,
  Alert,
  Linking,
} from "react-native"
import { useTranslation } from "react-i18next"

import { ENPermissionStatus } from "../PermissionsContext"
import { GlobalText } from "../components/GlobalText"
import { isPlatformiOS } from "../utils/index"

import { Layout, Spacing, Colors, Typography, Buttons } from "../styles"

interface HomeProps {
  enPermissionStatus: ENPermissionStatus
  requestPermission: () => void
}

const Home = ({
  enPermissionStatus,
  requestPermission,
}: HomeProps): JSX.Element => {
  const { t } = useTranslation()
  const [authorization, enablement] = enPermissionStatus
  const isEnabled = enablement === "ENABLED"
  const isAuthorized = authorization === "AUTHORIZED"

  const isEnabledAndAuthorized = isEnabled && isAuthorized

  const headerText = isEnabledAndAuthorized
    ? t("home.bluetooth.all_services_on_header")
    : t("home.bluetooth.tracing_off_header")
  const subheaderText = isEnabledAndAuthorized
    ? t("home.bluetooth.all_services_on_subheader")
    : t("home.bluetooth.tracing_off_subheader")
  const buttonText = t("home.bluetooth.tracing_off_button")

  const showUnauthorizedAlert = () => {
    Alert.alert(
      t("home.bluetooth.unauthorized_error_title"),
      t("home.bluetooth.unauthorized_error_message"),
      [
        {
          text: t("common.settings"),
          onPress: () => Linking.openSettings(),
        },
      ],
    )
  }

  const handleRequestPermission = () => {
    if (isAuthorized) {
      requestPermission()
    } else if (isPlatformiOS()) {
      showUnauthorizedAlert()
    }
  }

  return (
    <View style={style.container}>
      <View style={style.contentContainer}>
        <GlobalText style={style.headerText} testID={"home-header"}>
          {headerText}
        </GlobalText>
        <GlobalText style={style.subheaderText} testID={"home-subheader"}>
          {subheaderText}
        </GlobalText>
      </View>
      {!isEnabledAndAuthorized ? (
        <TouchableOpacity
          testID={"home-request-permissions-button"}
          onPress={handleRequestPermission}
          style={style.button}
        >
          <GlobalText style={style.buttonText}>{buttonText}</GlobalText>
        </TouchableOpacity>
      ) : null}
    </View>
  )
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "flex-end",
  },
  contentContainer: {
    position: "absolute",
    top: Layout.screenHeight / 3,
    width: "100%",
  },
  headerText: {
    ...Typography.largerFont,
    ...Typography.bold,
    lineHeight: Typography.mediumLineHeight,
    color: Colors.white,
    textAlign: "center",
  },
  subheaderText: {
    ...Typography.header4,
    textAlign: "center",
    color: Colors.white,
    marginTop: Spacing.medium,
  },
  button: {
    ...Buttons.largeWhite,
    width: "100%",
  },
  buttonText: {
    ...Typography.buttonTextDark,
  },
})

export default Home
