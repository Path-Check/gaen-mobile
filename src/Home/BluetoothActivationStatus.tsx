import React, { FunctionComponent } from "react"
import { Alert } from "react-native"
import { useSystemServicesContext } from "../SystemServicesContext"
import { ActivationStatus } from "./ActivationStatus"
import { useTranslation } from "react-i18next"
import { useNavigation } from "@react-navigation/native"
import { HomeStackScreens } from "../navigation"
import { openAppSettings } from "../gaen/nativeModule"

export const BluetoothActivationStatus: FunctionComponent = () => {
  const { t } = useTranslation()
  const navigation = useNavigation()
  const { isBluetoothOn } = useSystemServicesContext()

  const handleOnPressFix = () => {
    showFixBluetoothAlert()
  }

  const handleOnPressShowInfo = () => {
    navigation.navigate(HomeStackScreens.BluetoothInfo)
  }

  const showFixBluetoothAlert = () => {
    Alert.alert(
      t("home.bluetooth.bluetooth_disabled_error_title"),
      t("home.bluetooth.bluetooth_disabled_error_message"),
      [
        {
          text: t("common.back"),
          style: "cancel",
        },
        {
          text: t("common.settings"),
          onPress: () => openAppSettings(),
        },
      ],
    )
  }
  return (
    <ActivationStatus
      headerText={t("home.bluetooth.bluetooth_header")}
      isActive={isBluetoothOn}
      infoAction={handleOnPressShowInfo}
      fixAction={handleOnPressFix}
      testID={"home-bluetooth-status-container"}
    />
  )
}
