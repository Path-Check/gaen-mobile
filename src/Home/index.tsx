import React from "react"
import { StyleSheet, ImageBackground, View } from "react-native"
import { SvgXml } from "react-native-svg"
import { useTranslation } from "react-i18next"

import { usePermissionsContext } from "../PermissionsContext"
import { useStatusBarEffect } from "../navigation"
import Home from "./Home"

import { Icons, Images } from "../assets"
import { Spacing, Layout } from "../styles"

const HomeScreen = (): JSX.Element => {
  useStatusBarEffect("light-content")
  const { exposureNotifications } = usePermissionsContext()
  const { t } = useTranslation()

  return (
    <ImageBackground
      style={styles.backgroundImage}
      source={Images.BlueGradientBackground}
    >
      <View style={styles.iconContainer}>
        <SvgXml
          xml={Icons.StateNoContact}
          accessible
          accessibilityLabel={t("label.check_icon")}
          width={2 * Layout.screenWidth}
          height={2 * Layout.screenHeight}
        />
      </View>
      <View style={styles.homeContainer}>
        <Home
          enPermissionStatus={exposureNotifications.status}
          requestPermission={exposureNotifications.request}
        />
      </View>
    </ImageBackground>
  )
}

const styles = StyleSheet.create({
  backgroundImage: {
    flex: 1,
    height: "100%",
    width: "100%",
  },
  iconContainer: {
    position: "absolute",
    left: -(Layout.screenWidth / 2),
    bottom: -(Layout.screenHeight / 3),
  },
  homeContainer: {
    flex: 1,
    padding: Spacing.medium,
  },
})

export default HomeScreen
