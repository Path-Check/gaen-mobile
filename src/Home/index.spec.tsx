import React from "react"
import { Share } from "react-native"
import { render, fireEvent } from "@testing-library/react-native"
import { useNavigation } from "@react-navigation/native"
import "@testing-library/jest-native/extend-expect"

import Home from "./index"
import { HomeStackScreens } from "../navigation"
import { factories } from "../factories"
import { ConfigurationContext } from "../ConfigurationContext"
import { PermissionsContext } from "../Device/PermissionsContext"

jest.mock("@react-navigation/native")

const mockedApplicationName = "applicationName"
jest.mock("../Device/useApplicationInfo", () => {
  return {
    useApplicationName: () => {
      return {
        applicationName: mockedApplicationName,
      }
    },
  }
})

describe("Home", () => {
  it("allows the user to get more information on Exposure Detection status", () => {
    const navigateSpy = jest.fn()
    ;(useNavigation as jest.Mock).mockReturnValue({
      navigate: navigateSpy,
    })

    const { getByTestId } = render(<Home />)

    fireEvent.press(getByTestId("exposure-scanning-status-button"))
    expect(navigateSpy).toHaveBeenCalledWith(
      HomeStackScreens.ExposureDetectionStatus,
    )
  })

  it("allows users to share the application", () => {
    const configuration = factories.configurationContext.build()
    Share.share = jest.fn()

    const shareSpy = jest.spyOn(Share, "share")

    const { getByLabelText } = render(
      <ConfigurationContext.Provider value={configuration}>
        <Home />
      </ConfigurationContext.Provider>,
    )

    fireEvent.press(getByLabelText(`Share ${mockedApplicationName}`))

    expect(shareSpy).toHaveBeenCalledWith({
      message: `Check out this app ${mockedApplicationName}, which can help us contain COVID-19! ${configuration.appDownloadUrl}`,
    })
  })

  describe("When the exposure notification status is active", () => {
    it("renders an on message", () => {
      const permissionProviderValue = factories.permissionsContext.build({
        exposureNotifications: { status: "Active" },
      })

      const { getByText } = render(
        <PermissionsContext.Provider value={permissionProviderValue}>
          <Home />
        </PermissionsContext.Provider>,
      )

      expect(getByText("Exposure Detection On")).toBeDefined()
    })
  })

  describe("When the exposure notifications status is not active because location is off", () => {
    it("renders an off message", () => {
      const permissionProviderValue = factories.permissionsContext.build({
        exposureNotifications: { status: "LocationOffAndRequired" },
      })

      const { getByText } = render(
        <PermissionsContext.Provider value={permissionProviderValue}>
          <Home />
        </PermissionsContext.Provider>,
      )

      expect(getByText("Exposure Detection Off")).toBeDefined()
    })
  })

  describe("When exposure notifications are disabled", () => {
    it("renders an off message", () => {
      const permissionProviderValue = factories.permissionsContext.build({
        exposureNotifications: { status: "Disabled" },
      })

      const { getByText } = render(
        <PermissionsContext.Provider value={permissionProviderValue}>
          <Home />
        </PermissionsContext.Provider>,
      )

      expect(getByText("Exposure Detection Off")).toBeDefined()
    })
  })

  describe("When exposure notifications are unauthorized", () => {
    it("renders an off message", () => {
      const permissionProviderValue = factories.permissionsContext.build({
        exposureNotifications: { status: "Unauthorized" },
      })

      const { getByText } = render(
        <PermissionsContext.Provider value={permissionProviderValue}>
          <Home />
        </PermissionsContext.Provider>,
      )

      expect(getByText("Exposure Detection Off")).toBeDefined()
    })
  })
})
