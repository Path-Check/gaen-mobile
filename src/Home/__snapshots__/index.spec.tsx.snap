// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/blueGradientBackground.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "bottom": -444.6666666666667,
          "left": -375,
          "position": "absolute",
        }
      }
    >
      <SvgXml
        accessibilityLabel="Checkmark icon"
        accessible={true}
        height={2668}
        width={1500}
        xml="<svg width=\\"850\\" height=\\"850\\" viewBox=\\"0 0 850 850\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<g opacity=\\"0.2\\">
<circle cx=\\"425\\" cy=\\"425\\" r=\\"85.5\\" stroke=\\"white\\"/>
<circle cx=\\"425\\" cy=\\"425\\" r=\\"149.5\\" stroke=\\"white\\"/>
<circle cx=\\"425\\" cy=\\"425\\" r=\\"217.5\\" stroke=\\"white\\"/>
<circle cx=\\"425\\" cy=\\"425\\" r=\\"283.5\\" stroke=\\"white\\"/>
<circle cx=\\"425\\" cy=\\"425\\" r=\\"352.5\\" stroke=\\"white\\"/>
<circle cx=\\"425\\" cy=\\"425\\" r=\\"424.5\\" stroke=\\"white\\"/>
</g>
<path d=\\"M425 465C447.091 465 465 447.091 465 425C465 402.909 447.091 385 425 385C402.909 385 385 402.909 385 425C385 447.091 402.909 465 425 465Z\\" fill=\\"white\\"/>
<path d=\\"M425 457C442.673 457 457 442.673 457 425C457 407.327 442.673 393 425 393C407.327 393 393 407.327 393 425C393 442.673 407.327 457 425 457Z\\" fill=\\"url(#paint0_linear)\\"/>
<path d=\\"M425 412C417.832 412 412 417.832 412 425C412 432.168 417.832 438 425 438C432.168 438 438 432.168 438 425C438 417.832 432.168 412 425 412ZM425 435.075C419.436 435.075 414.925 430.565 414.925 425C414.925 419.436 419.436 414.925 425 414.925C430.564 414.925 435.075 419.436 435.075 425C435.075 430.565 430.564 435.075 425 435.075Z\\" fill=\\"white\\"/>
<path d=\\"M430.844 419.727C430.219 419.215 429.298 419.305 428.786 419.93L423.15 426.797L421.24 424.334C420.745 423.695 419.826 423.58 419.188 424.074C418.549 424.57 418.433 425.488 418.928 426.127L421.962 430.039C422.234 430.391 422.653 430.599 423.097 430.605C423.104 430.606 423.111 430.606 423.118 430.606C423.556 430.606 423.97 430.409 424.248 430.071L431.046 421.785C431.559 421.161 431.468 420.239 430.844 419.727Z\\" fill=\\"white\\"/>
<defs>
<linearGradient id=\\"paint0_linear\\" x1=\\"425\\" y1=\\"361\\" x2=\\"361\\" y2=\\"425\\" gradientUnits=\\"userSpaceOnUse\\">
<stop stop-color=\\"#6979F8\\"/>
<stop offset=\\"1\\" stop-color=\\"#3A4CD7\\"/>
</linearGradient>
</defs>
</svg>
"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "position": "absolute",
              "top": 444.6666666666667,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexSans-Bold",
                  "fontSize": 22,
                  "fontWeight": "500",
                  "lineHeight": 24,
                  "textAlign": "center",
                },
              ]
            }
            testID="home-header"
          >
            Exposure Notifications Disabled
          </Text>
          <Text
            style={
              Array [
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 15,
                  "lineHeight": 20,
                  "marginTop": 20,
                  "textAlign": "center",
                },
              ]
            }
            testID="home-subheader"
          >
            Enable Exposure Notifications to receive information about possible exposures
          </Text>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderColor": "#ffffff",
              "borderRadius": 8,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 25,
              "paddingTop": 24,
              "width": "100%",
            }
          }
          testID="home-request-permissions-button"
        >
          <Text
            style={
              Array [
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#1f2c9b",
                  "fontSize": 19,
                  "fontWeight": "500",
                },
              ]
            }
          >
            Allow Exposure Notifications
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
